architecture/2024-10-user-defined-types-authoring-experience-feature-spec.md

```diff
@@ -0,0 +1,230 @@
+The key scenarios for UDTs are outlined [here](/architecture/2024-06-resource-extensibility-feature-spec.md). We will decompose these scenarios specific to the authoring experience for defining the schema for the User-Defined-Type.
+
+Deb, a platform engineer at a Financial Services company who is responsible for building the custom internal developer platform. He has started the journey to enable radification one of their Budgets app. The Budgets app relies on an internal messaging service called Plaid. Deb wants to use Plaid as a resource type in Radius to deploy the application seamlessly and leverage all the features of Radius, such as Recipes, Connections, and the rad CLI, with ease and flexibility.
+
+### Scenario 1: Deb defines the schema for the internal messaging service Plaid
+<!-- One or two sentence summary -->
+As called out in the [feature spec](/architecture/2024-06-resource-extensibility-feature-spec.md), since `yaml` is the most common and standard format across open-source communities, we want to enable users to define the schema in a custom `yaml` format.
+
+1. Deb creates a sample schema definition for the internal messaging service Plaid
+   
+   Below are the basic set of inputs needed required to author a schema definition
+    
+    |Input | Description | Required|
+    |------|-------------|---------|
+    | Resource namespace | The namespace for the resource provider | Yes |
+    | Resource type name | The name of the resource type | Yes |
+    | Resource type description | The description of the resource type | No |
+    | Properties | The data model of the resource type | Yes |
+    | API Version | The version of the resource type | Yes |
+
+    Deb creates a plaid.yaml file with the following schema definition
+
+    ```yaml
+    namespace: 'Mycompany.Messaging'
+    resourceTypes:
+        'Mycompany.Messaging/plaidResource':
+            name: 'Name of the resource'
+            version: '2024-10-01'
+                properties: 
+                    required: ['queueName','host', 'port', 'connectionString']
+                    queueName:
+                        type: 'string'
+                        description: 'Name of the queue'
+                    host:
+                        type: 'string'
+                        description: 'Hostname of the messaging service'        
+                    port:
+                        type: 'string'
+                        description: 'Port'
+                    connectionString:
+                        type: 'string'
+                        description: 'Connection string to the messaging service'
+    ....
+    ```
+ Questions
+1. Is resource namespace needed or can we parse this from the resource type ?
+
+ Now Deb can register this resource type via the CLI
+
+    ```bash
+    rad resource-type create -f plaid.yaml
```

Comment:
Is the file meant to container a single resource type or all types for a resource provider? It looks like it's doing both.

Reply:
I intended to do it RP level. updated the file structure and creation